client/js/components/show-hide.js

```diff
@@ -11,8 +11,24 @@ const showHide = (state) => {
   const init = () => {
     el.setAttribute('aria-haspopup', 'true');
     options.trigger.setAttribute('aria-expanded', getActive());
+
+    setDataTrackAria(getActive());
   };
 
+  function setDataTrackAria(value) {
+    const dataTrackOptions = options.trigger.getAttribute('data-track-click');
+
+    if (!dataTrackOptions) return;
+
+    const updatedTrackOptions = JSON.stringify(Object.assign(
+      {},
+      JSON.parse(dataTrackOptions),
+      {actionFromAriaExpanded: value ? 'open' : 'close'}
```

Comment:
Na, I prefer your solution that slipped past me, naming the action makes way more sense.